config/config.go

```diff
@@ -207,9 +207,35 @@ func (c *Config) UnmarshalYAML(unmarshal func(interface{}) error) error {
 		}
 		names[rcv.Name] = struct{}{}
 	}
+
+	// The root route must not have any matchers as it is the fallback node
+	// for all alerts.
+	if len(c.Route.Match) > 0 || len(c.Route.MatchRE) > 0 {
```

Comment:
c.Route could be null, we should check for that first or change it not to be a pointer